Spurious, positionless overrides-nullary-method warning when extending under separate compilation

Scala version: 2.13.11

### T.scala

```scala
trait T1 {
  def method: Int
}
trait T2 extends T1 {
  def method() = 0
}
```

### C.scala

```scala
class C extends T2
```

### Reproduction steps

```
scalac T.scala
scalac -classpath . C.scala
```

### Result

compiling `T.scala` produces a correct warning:

```
T.scala:5: warning: method with a single empty parameter list overrides method without any parameter list
  def method() = 0
      ^
1 warning
```

but then compiling `C.scala` produces this:

```
warning: method with a single empty parameter list overrides method without any parameter list
1 warning
```

we shouldn't be issuing this warning at all IMO, but also note the lack of position information, which means it's difficult to figure out where it's even coming from

## Context

This came up over at https://github.com/scala/scala/pull/10484 , where after a recent change for 2.13.12 the attempt to issue the warning actually crashes the compiler.

```
➜ sandbox scv 2.13.12-bin-fa1c9eb A.scala B.scala
A.scala:5: warning: method with a single empty parameter list overrides method without any parameter list
  def method() = 0
      ^
1 warning


➜ sandbox scv 2.13.12-bin-910c59d A.scala B.scala
A.scala:5: warning: method with a single empty parameter list overrides method without any parameter list [quickfixable]
  def method() = 0
      ^
A.scala:5: warning: method with a single empty parameter list overrides method without any parameter list
  def method() = 0
      ^
2 warnings
```

so the second warning started showing up with https://github.com/scala/scala/pull/10484.

I think before we only got a single warning because the second one was filtered by the `FilteringReporter`. After that PR, the two warnings no longer have the same message because one has `[quickfixable]`, but not the other. So they are both shown.

--- a/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala
@@ -461,7 +461,8 @@ abstract class RefChecks extends Transform {
               }
               else if (other.paramss.isEmpty && !member.paramss.isEmpty &&
                 !javaDetermined(member) && !member.overrides.exists(javaDetermined) &&
-                !member.hasAnnotation(BeanPropertyAttr) && !member.hasAnnotation(BooleanBeanPropertyAttr)
+                !member.hasAnnotation(BeanPropertyAttr) && !member.hasAnnotation(BooleanBeanPropertyAttr) &&
+                member.pos.isDefined  // scala/bug#12851
               ) {
                 val msg = "method with a single empty parameter list overrides method without any parameter list"
                 val namePos = member.pos.focus.withEnd(member.pos.point + member.decodedName.length)

--- a/test/files/pos/t12851/C_2.scala
+++ b/test/files/pos/t12851/C_2.scala
@@ -1,0 +1,2 @@
+// scalac: -Werror
+class C extends T2

--- a/test/files/pos/t12851/T_1.scala
+++ b/test/files/pos/t12851/T_1.scala
@@ -1,0 +1,7 @@
+
+trait T1 {
+  def f: Int
+}
+trait T2 extends T1 {
+  def f() = 42
+}

